TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Utility.php

```diff
@@ -44,6 +47,11 @@ public static function renderValidNodeName($name)
         // Ensure only allowed characters are left
         $name = preg_replace('/[^a-z0-9\-]/', '', $name);
 
+        // Make sure we don't have an empty string left.
+        if ($name === '') {
+            $name = 'node-' . strtolower(md5($originalName));
```

Comment:
but no need for the full md5 for that tho, 13 characters would suffice for that and keep it consistent with the convention